src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadAsyncState.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Threading;
+
+namespace System.Text.Json.Serialization
+{
+    internal sealed class ReadAsyncState : IDisposable
```

Comment:
Can this be a struct?

Reply:
Per the other thread, I think it can be removed altogether.